src/Eagle.ts

```diff
@@ -763,6 +763,47 @@ export class Eagle {
         return false;
     }
 
+    getOutermostSelectedNodes = () : Node[] => {
+        const outermostNodes : Node[] = []
+        const selectedNodes = this.selectedObjects()
+        const eagle = this
+
+        selectedNodes.forEach(function(object){
+            if(object instanceof Node){
```

Comment:
Maybe rather than nest most of the function inside this if, you could invert it as an early exit.

if (!(object instanceof Node)){
    return outermostNodes;
}

then continue with the rest of the function